ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1532,9 +1532,12 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
           getOptions().ageAtAccessTracker.trackAgeAtAccess(serverBlobInfo.getBlobProperties().getCreationTimeInMs());
           blobData = blobAll.getBlobData();
           encryptionKey = blobAll.getBlobEncryptionKey();
-          if (encryptionKey == null) {
-            // set blobInfo only if decryption is not required. If not, maybeProcessCallbackAndComplete() will set appropriate
-            // value
+          if (encryptionKey == null || options.getChunkIdsOnly) {
+            // set blobInfo only if decryption is not required, or if its a chunk id only request. This is because
+            // blobInfo has user metadata which is encrypted for encrypted blobs. When decryption is required,
+            // maybeProcessCallbackAndComplete() will set appropriate value after decryption.
+            // Note that in case of getChunkIdsOnly call, we don't need the user metadata but still might need blob
+            // properties.
```

Comment:
nit: It's better to include the reason of why we need the blob properties for future reference.

Reply:
done.